cli/src/workspace_import.rs

```diff
@@ -0,0 +1,78 @@
+use std::{path::PathBuf, vec};
+
+use libparsec::{FsPath, OpenOptions, VlobID};
+use tokio::io::AsyncReadExt;
+
+use crate::utils::load_client;
+
+crate::clap_parser_with_shared_opts_builder!(
+    #[with = config_dir, device, password_stdin]
+    pub struct WorkspaceImport {
+        /// Workspace id
+        #[arg(short, long, value_parser = VlobID::from_hex)]
+        workspace_id: VlobID,
+        /// Local file to copy
+        src: PathBuf,
+        /// Workspace destination path
+        dest: FsPath,
+    }
+);
+
+pub async fn workspace_import(args: WorkspaceImport) -> anyhow::Result<()> {
+    let WorkspaceImport {
+        src,
+        dest,
+        workspace_id,
+        password_stdin,
+        device,
+        config_dir,
+    } = args;
+
+    log::trace!(
+        "workspace_import: {src} -> {workspace_id}:{dst}",
+        src = src.display(),
+        dst = dest
+    );
+
+    let client = load_client(&config_dir, device, password_stdin).await?;
+    let workspace = client.start_workspace(workspace_id).await?;
+    let fd = workspace
+        .open_file(
+            dest,
+            OpenOptions {
+                read: false,
+                write: true,
+                truncate: true,
+                create: true,
+                create_new: false,
+            },
+        )
+        .await?;
+
+    let file = tokio::fs::File::open(&src).await?;
+    let mut buf_file = tokio::io::BufReader::new(file);
+    let mut buffer = vec![0_u8; 4096];
+    let mut dst_offset = 0_usize;
+
+    log::debug!("Copying file to workspace");
+    loop {
+        let bytes_read = buf_file.read(&mut buffer).await?;
+
+        if bytes_read == 0 {
+            break;
+        }
+
+        let mut bytes_written = 0_usize;
+        while bytes_written < bytes_read {
+            bytes_written = workspace
+                .fd_write(fd, dst_offset as u64, &buffer[bytes_written..bytes_read])
```

Comment:
`fd_write` always write the whole buffer given to it

I realize returning the size of data written is confusing here, so I think we should just return an empty tuple instead (I've create an issue about this)

Reply:
If that the case, I propose to rename `fd_write` to `fd_write_all` to be similar to [`Write::write_all`](https://doc.rust-lang.org/std/io/trait.Write.html#method.write_all)